modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -249,66 +238,114 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
 
     _build_mesh = &addMeshSubgenerator("BlockDeletionGenerator", name() + "_deleted", params);
   }
+  // Remove outer assembly sidesets created during assembly generation
+  {
+    // Get outer boundaries of all constituent assemblies based on assembly_type,
+    // skipping all dummy assemblies
+    std::vector<BoundaryName> boundaries_to_delete = {};
+    for (const auto & pattern_x : _pattern)
+    {
+      for (const auto & pattern_idx : pattern_x)
+      {
+        const auto assembly_name = _inputs[pattern_idx];
+        if (assembly_name == _empty_key)
+          continue;
+        const auto assembly_id = getMeshProperty<subdomain_id_type>("assembly_type", assembly_name);
+        const BoundaryName boundary_name = "outer_assembly_" + std::to_string(assembly_id);
+        if (std::find(boundaries_to_delete.begin(), boundaries_to_delete.end(), boundary_name) ==
```

Comment:
if you change the equivalent code in AssemblyMG to a set or something else, dont forget this one